### PR TITLE
[RHACS] [4.5 and later versions] ROX-24749: Updating Deleting global resources documentation - Uninstall section

### PR DESCRIPTION
**Pull request by agantony**
Version(s):
4.5+

Issue:
https://issues.redhat.com/browse/ROX-24749

Link to docs preview:
https://82260--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/uninstall-acs.html#delete-acs-global-resources_uninstall-acs

SME review:
- [x] SME has approved this change.

Additional information:
- Merge into `rhacs-docs-main` and cherry-pick to:
    - `rhacs-docs-4.6`
    - `rhacs-docs-4.5`

### CODE REVIEW

**Review by vikin91 — APPROVED** [review]
I reviewed the changes and they look good!

The commands itself were not part of this PR, but the fragment `grep stackrox | xargs kubectl delete --wait` is not a great selector of things to delete. It may delete something that has `stackrox` word in its name but is not installed by us. Not sure whether that is a real problem...

**Comment by agantony** [comment]
> I reviewed the changes and they look good!
> 
> The commands itself were not part of this PR, but the fragment `grep stackrox | xargs kubectl delete --wait` is not a great selector of things to delete. It may delete something that has `stackrox` word in its name but is not installed by us. Not sure whether that is a real problem...

@vikin91 I would appreciate it if you could suggest a better option or direct me to someone who could help us with this? Thx!

**Comment by vikin91** [comment]
> > I reviewed the changes and they look good!
> > The commands itself were not part of this PR, but the fragment `grep stackrox | xargs kubectl delete --wait` is not a great selector of things to delete. It may delete something that has `stackrox` word in its name but is not installed by us. Not sure whether that is a real problem...
> 
> @vikin91 I would appreciate it if you could suggest a better command option or direct me to someone who could help us with this? Thx!

I am afraid, this would need some time to find a better option. We would need to analyze the set of labels and maybe use them as selectors. I think the PR can be merged as it is as we are addressing here a different issue. We can open a follow-up and improve the command.
Maybe the ACS install team would be a better source of information for this command?

**Comment by agantony** [comment]
> > > I reviewed the changes and they look good!
> > > The commands itself were not part of this PR, but the fragment `grep stackrox | xargs kubectl delete --wait` is not a great selector of things to delete. It may delete something that has `stackrox` word in its name but is not installed by us. Not sure whether that is a real problem...
> > 
> > 
> > @vikin91 I would appreciate it if you could suggest a better command option or direct me to someone who could help us with this? Thx!
> 
> I am afraid, this would need some time to find a better option. We would need to analyze the set of labels and maybe use them as selectors. I think the PR can be merged as it is as we are addressing a different issue here. We can open a follow-up and improve the command - would that be okay? Maybe the ACS install team would be a better source of information for this command?

Noted thanks, I created a separate Jira task to handle this.
https://issues.redhat.com/browse/ROX-26392

File: modules/delete-acs-global-resources.adoc [hunk 1]
```diff
@@ -1,39 +1,66 @@
...
+.. Retrieve all the StackRox-related cluster roles, cluster role bindings, roles, role bindings, and PSPs, and then delete them by running the following command:
++ 
 [source,terminal]
 ----
 $ oc get clusterrole,clusterrolebinding,role,rolebinding,psp -o name | grep stackrox | xargs oc delete --wait
 ----
 +
+[NOTE]
+====
+You might receive the `error: the server doesn't have a resource type "psp"` error message in {product-title-short} 4.4 and later versions because the pod security policies (PSPs) are deprecated and removed from Kubernetes in version 1.25, except for clusters with older Kubernetes versions.
```
Comment (jeana-redhat) [inline modules/delete-acs-global-resources.adoc:25]: Consider breaking this long sentence up for a bit of an easier read :)

File: modules/delete-acs-global-resources.adoc [hunk 2]
```diff
@@ -1,39 +1,66 @@
...
+====
+
+.. Delete all the custom security context constraints (SCCs) labeled with `app.kubernetes.io/name=stackrox` by running the following command:
```
Comment (jeana-redhat) [inline modules/delete-acs-global-resources.adoc:28]:
```suggestion
.. Delete the custom security context constraints (SCCs) that are labeled with `app.kubernetes.io/name=stackrox` by running the following command:
```

File: modules/delete-acs-global-resources.adoc [hunk 3]
```diff
@@ -1,39 +1,66 @@
...
 ----
 $ oc delete scc -l "app.kubernetes.io/name=stackrox"
 ----
 +
+[NOTE]
+====
+You might receive the `No resources found` error message in {product-title-short} 4.4 and later versions because the custom SCCs with this label are no longer used in these versions.
+====
+
+.. Delete the `ValidatingWebhookConfiguration` named `stackrox` by running the following command:
```
Comment (jeana-redhat) [inline modules/delete-acs-global-resources.adoc:40]: Could one of these work?

```suggestion
.. Delete the `ValidatingWebhookConfiguration` object named `stackrox` by running the following command:
```

```suggestion
.. Delete the validating webhook configuration named `stackrox` by running the following command:
```

```suggestion
.. Delete the StakRox `ValidatingWebhookConfiguration` object by running the following command:
```

```suggestion
.. Delete the StakRox validating webhook configuration by running the following command:
```

File: modules/delete-acs-global-resources.adoc [hunk 4]
```diff
@@ -1,39 +1,66 @@
...
+.. Retrieve all the StackRox-related cluster roles, cluster role bindings, roles, role bindings, and PSPs, and then delete them by running the following command:
 +
 [source,terminal]
 ----
 $ kubectl get clusterrole,clusterrolebinding,role,rolebinding,psp -o name | grep stackrox | xargs kubectl delete --wait
 ----
 +
+[NOTE]
+====
+You might receive the `error: the server doesn't have a resource type "psp"` error message in {product-title-short} 4.4 and later versions because the pod security policies (PSPs) are deprecated and no longer available, except for clusters with older Kubernetes versions.
```
Comment (jeana-redhat) [inline modules/delete-acs-global-resources.adoc:58]: same comment here as earlier

File: modules/delete-acs-global-resources.adoc [hunk 5]
```diff
@@ -1,39 +1,66 @@
...
+====
+
+.. Delete the `ValidatingWebhookConfiguration` named `stackrox` by running the following command:
```
Comment (jeana-redhat) [inline modules/delete-acs-global-resources.adoc:61]: same comment here as earlier

**Review by jeana-redhat — COMMENTED** [review]
Just a few suggestions here.

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

One thing to consider (later):

**Procedure with both CLIs**

* steps for OCP
  1. CLI step
* steps for k8s
  1. CLI step

I can see why this approach made sense for the other two procedures on this page - they each are single step procedures. But once it starts to get longer, like this one, the structure becomes a little harder to parse easily.

I would suggest separating these out. That's out of scope for this PR, because you would want to do it for the other procedures as well to keep everything consistent. 

But in general, if we are thinking about the user experience, a user is probably not going to be switching between CLIs. So it makes sense to show this process start to finish using OCP and start to finish using k8s as different user stories.

File: modules/delete-acs-global-resources.adoc [hunk 6]
```diff
@@ -1,39 +1,66 @@
...
 //
 // * installing/uninstall-acs.adoc
+
 :_mod-docs-content-type: PROCEDURE
 [id="delete-acs-global-resources_{context}"]
 = Deleting global resources
 
 [role="_abstract"]
-You can delete the global resources that {product-title} creates, by using the {ocp} or Kubernetes command-line interface.
+You can delete the global resources that {rh-rhacs-first} creates by using the {ocp} or Kubernetes command-line interface (CLI).
```
Comment (skrthomas) [inline modules/delete-acs-global-resources.adoc:10]:
```suggestion
You can delete the global resources that {rh-rhacs-first} creates by using the {oc-first} or Kubernetes command-line interface (CLI).
```

Comment (skrthomas) [reply modules/delete-acs-global-resources.adoc:10]: For mentions of the OpenShift CLI (`oc`), use the attribute {oc-first}.

File: modules/delete-acs-global-resources.adoc [hunk 7]
```diff
@@ -1,39 +1,66 @@
...
 
 .Procedure
-* Delete global resources:
-** On {ocp}:
-+
+
+* To delete the global resources by using the {ocp} CLI, perform the following steps:
```
Comment (skrthomas) [inline modules/delete-acs-global-resources.adoc:14]:
```suggestion
* To delete the global resources by using the {oc-first}, perform the following steps:
```

File: modules/delete-acs-global-resources.adoc (hunk 6 above)
Comment (skrthomas) [reply modules/delete-acs-global-resources.adoc:10]: I guess this reads fine as is and I don't think its mandatory to necessarily put it like this so I won't block merge over it. In the SSG, I see OpenShift CLI is approved. But I think its preferable to use this attribute, so you get the `oc` in parentheses.

**Review by skrthomas — COMMENTED** [review]
A couple of nits but not worth blocking merge over

**Pull request merged**